### PR TITLE
🎉 Implement 721 to 3525 extension

### LINKED ISSUE #1: Implement 721 to 3525 extension

### PR DESCRIPTION
**Pull request by bal7hazar**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+<div align="center">
+  <h1 align="center">ERC-4906: EIP-721 Metadata Update Extension</h1>
```
Comment (tekkac) [inline README.md]: Not really up to date?

File: src/migrator/presets/Migrator.cairo [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+@view
+func slot{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (slot: Uint256) {
+    let (slot) = Migrator.slot();
+    return (slot=slot);
+}
+
+@view
+func value{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (value: Uint256) {
```
Comment (tekkac) [inline src/migrator/presets/Migrator.cairo:49]: Only one predefined value? Can we instead handle different sizing?

File: src/migrator/library.cairo [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        // [Interaction] Transfer token to the contract
+        let (source_address) = Migrator_source_address_.read();
+        let (caller) = get_caller_address();
+        let (contract_address) = get_contract_address();
+        IERC3525Full.transferFrom(
+            contract_address=source_address, from_=caller, to=contract_address, tokenId=token_id
+        );
+
+        // [Interaction] Burn the token
+        IERC3525Full.burn(contract_address=source_address, tokenId=token_id);
```
Comment (tekkac) [inline src/migrator/library.cairo]: Shouldn't this be IERC721 instead? (same on line 137)

File: src/migrator/library.cairo [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+func Migrator_value_() -> (value: Uint256) {
+}
+
+namespace Migrator {
+    //
+    // Constructor
+    //
+
+    func initializer{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        source_address: felt, target_address: felt, slot: Uint256, value: Uint256
```
Comment (tekkac) [inline src/migrator/library.cairo:58]: Since the source and target are fixed here, shouldn't we also check they support the wanted ERC165 interfaces?

**Review by tekkac — CHANGES_REQUESTED** [review]
For our purposes, we might need to test that we can migrate and farm in one tx using muticall.
Right now, I'm not sure if that is possible as it would require to know the `new_token_id` in advance.

File: src/migrator/presets/Migrator.cairo (hunk 2 above)
Comment (bal7hazar) [reply src/migrator/presets/Migrator.cairo:49]: Not needed since we will apply the migrator extension to all minters individually

**Review by tekkac — APPROVED** [review]
LGTM

**Pull request merged**